hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyDeleteRequest.java

```diff
@@ -117,12 +118,17 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       acquiredLock = omMetadataManager.getLock().acquireLock(BUCKET_LOCK,
           volumeName, bucketName);
 
-      // Not doing bucket/volume checks here. In this way we can avoid db
-      // checks for them.
-      // TODO: Once we have volume/bucket full cache, we can add
-      // them back, as these checks will be inexpensive at that time.
-      OmKeyInfo omKeyInfo = omMetadataManager.getKeyTable().get(objectKey);
+      // Check bucket/volume exist.
+      String volumeKey = omMetadataManager.getVolumeKey(volumeName);
+      String bucketKey = omMetadataManager.getBucketKey(volumeName, bucketName);
+      if (!omMetadataManager.getBucketTable().isExist(bucketKey)) {
```

Comment:
We can move this to a base class method, which can be used every where.

Reply:
Thanks @bharatviswa504 
That's my fault to ignore the base class method.
I am going to update the issue.